renv.lock

```diff
@@ -0,0 +1,566 @@
+{
```

Comment:
package `readr` is not captured in `renv.lock`. Due to no `readr` package found, loading fill in tab#3 will cause disconnect from server. I guess it is because we were calling `readr` using `readr::read_csv`, but never load the library at the beginning, like `library(readr)`. And the `renv` will not capture the packages that are not loaded using `library()`?

Reply:
Hm, rather than `library()` it's more about having used one of the existing tools in R (`install.packages()`, `remove.packages()`) to deal with packages. If you snapshot `renv`, with `renv::snapshot()` after running an `install.packages("library_name")` it will update the lockfile with that package.

As a recommended "best" practice, do you think it's a good idea to avoid accessing library methods with `::` and swtich to loading libraries and accessing their methods normally? I see it's the same case with `reticulate`, `jsonlite` and `DT`. What is your opinion on this design @rrchai?

Reply:
Or we can load the library in this PR and aim to remove the `::` in the next round of refactor.

Reply:
>As a recommended "best" practice, do you think it's a good idea to avoid accessing library methods with :: and swtich to loading libraries and accessing their methods normally?

I would say yes and no. I could definitely only use `library()` to use all packages. However, in R, while working for many packages it may be chances to conflict one package with another. So just using `library()` could be really dependent on the order of how load libraries.

For example, `dplyr::filter()` vs `stats::filter()` and `shinydashboard::dashboardPage()` vs `shinydashboardPlus::dashboardPage()`. Some function names are overlapped in different packages.

1. If we would like to only use `library()` and avoid `::` to use `dashboardPage()` function from `shinydashboardPlus` package. We need to load `shinydashboardPlus` after `shinydashboard`. So R will use the latest loaded package's` dashboardPage()` if they have conflicts with other packages.
2. In my opinion, I would think using `::` is a better approach, so we don't need to worry about the order of loading libraries. Also, it is a reminder for us what package the function is coming from. As far as I know, R doesn't have ability to rename the loaded libraries, like `import xxx as xxx` in python.

I agree that we should normally load used libraries using `library()` at the beginning, since just using `::` to call library doesn't seem like the best practice. If there is no conflict, we could remove the unnecessary `::`, such as `reticulate::` and `readr::read_csv` in the next round of refactor.

Back to the issue, as you suggest, installing the `readr` and then updating `renv.lock` can solve the issue for now. 
- I am little curious why the `readr` was not captured when we used `renv::snapshot()` in the first place. Because if `readr` was not installed, then `readr::` should not work, either.

Reply:
Thank you for the explanation @rrchai. I agree that `::` helps resolve conflicts when there are methods with the same name in different libraries.

> I am little curious why the readr was not captured when we used renv::snapshot() in the first place. Because if readr was not installed, then readr:: should not work, either.

Oh so the first time when transitioning to the use of `renv` (this PR), I installed all the packages manually based on what libraries were being loaded using `library()`, and forgot to do it for the ones that were being loaded with the `::`. That's how `readr` wasn't updated in the `renv.lock` file. I'll go ahead and update the `renv.lock` file and make sure `readr` is in there. I'll make sure I haven't missed any other packages in the same way. And you can test it after on that.

Reply:
That will be great! Thanks @sujaypatil96